packages/clay-form/src/MultiSelect.tsx

```diff
@@ -0,0 +1,314 @@
+	forwardRef,
+	helpText,
+	items,
+	label,
+	onItemsChange,
+	spritemap,
+	validateOnBlur = false,
+	validator = () => true,
+	inputValue = '',
+	onInputChange,
+	sourceItems,
+	...otherProps
+}) => {
+	const inputRef = useRef<HTMLInputElement>(null);
+	const lastItemRef = useRef<HTMLSpanElement>(null);
+	const [active, setActive] = useState(false);
+	const [isFocused, setIsFocused] = useState();
+	const [isValid, setIsValid] = useState(true);
+
+	useLayoutEffect(() => {
+		if (sourceItems) {
+			const matchedItems = sourceItems.filter(
+				item => inputValue && item.match(inputValue)
+			);
+
+			setActive(matchedItems.length !== 0);
+		}
+	}, [sourceItems, inputValue]);
+
+	const setNewValue = (newVal: string) => {
+		if (validator(newVal)) {
+			onItemsChange([...items, newVal]);
+
+			if (onInputChange) {
+				onInputChange('');
+			}
+		} else {
+			setIsValid(false);
+		}
+	};
+
+	const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+		const {keyCode} = event;
+
+		if (inputValue && DELIMITER_KEYS.includes(keyCode)) {
+			event.preventDefault();
+
+			setNewValue(inputValue);
+		} else if (
+			!inputValue &&
+			keyCode === BACKSPACE_KEY &&
+			inputRef.current &&
+			lastItemRef.current
+		) {
+			inputRef.current.blur();
+			lastItemRef.current.focus();
+		} else {
+			setIsValid(true);
+		}
+	};
+
+	const handleBlur = (event: React.FocusEvent<HTMLInputElement>) => {
+		const {value} = event.target;
+
+		setIsFocused(false);
+
+		if (value && validateOnBlur) {
+			setNewValue(value);
+		}
+	};
+
+	const handlePaste = (event: React.ClipboardEvent<HTMLInputElement>) => {
+		const pastedText = event.clipboardData.getData('Text');
+
+		const pastedItems = pastedText
+			.split(',')
+			.map(item => item.trim())
+			.filter(Boolean);
+
+		if (pastedItems.length > 0) {
+			event.preventDefault();
+
+			const validItems = pastedItems.filter(
+				item => item && validator(item)
+			);
+
+			if (validItems.length) {
+				onItemsChange([...items, ...validItems]);
+
+				if (validItems.length === items.length) {
+					setIsValid(false);
+				}
+			}
+		}
+	};
+
+	return (
+		<div className="form-group" ref={forwardRef}>
+			{label && <label>{label}</label>}
+
+			<div className="input-group input-group-stacked-sm-down">
+				<div
+					className={classNames('input-group-item', {
+						focus: isFocused && isValid,
+						['has-error']: !isValid,
+					})}
+				>
+					<div className="form-control form-control-tag-group">
+						{items &&
+							items.length > 0 &&
+							items.map((item, i) => {
+								const removeItem = () =>
+									onItemsChange([
+										...items.slice(0, i),
+										...items.slice(i + 1),
+									]);
+
+								return (
+									<ClayLabel
+										closeButtonProps={{
+											onClick: removeItem,
+										}}
+										key={i}
+										onKeyDown={() => {
+											if (inputRef.current) {
+												inputRef.current.focus();
+											}
+
+											removeItem();
```

Comment:
The functionality I am aiming for is..
 
 if input is blank and user hits backspace -> last label is focused
 if the label is focused and user hits "backspace" -> label is deleted
 if the label's "x" is focused and user hits "enter" -> label is deleted
user should be able to tab through labels and "x" icons

Does that seem right to you? Focusing the label and hitting "enter" seemed weird to me that it would delete.

Reply:
Hm I do not know exactly, but maybe the focus could just stay on the Label and not have the focus on `x`. In the previous version the button was getting `tabindex ="0"`.

Reply:
> Focusing the label and hitting "enter" seemed weird to me that it would delete.

Also seemed a bit counter-intuitive to me.